sycl/test-e2e/Basic/launch_queries/max_num_work_groups.cpp

```diff
@@ -187,7 +189,12 @@ int test_max_num_work_groups(sycl::queue &q, const sycl::device &dev) {
        } else {
          cgh.parallel_for(launch_range, KernelName{acc});
        }
-     }).wait();
+     }).wait_and_throw();
+    if constexpr (KernelName::HasLocalMemory)
+      std::cerr << "Test (LocalMemory) with exceeded resource limits failed\n";
+    else
+      std::cerr << "Test with exceed resource limits failed\n";
+    return 1; // We shouldn't be here, exception is expected
```

Comment:
Alternatively, you could change the return after the `catch` block to `1`. With this, it is now dead code.

Reply:
Sure. Addressed in a new commit.
Apologies for the force-push, it was because I rebased on latest `sycl` branch and messed my branch updating strategy here with what I normally do in unified-runtime where I rebase on top rather the merge.